rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py

```diff
@@ -0,0 +1,719 @@
+        #    results["DISAGREE_intrinsic_rewards"] = tf.reduce_mean(
+        #        dream_data["rewards_intrinsic_t1_to_H_B"]
+        #    )
+
+        # Add world model loss results.
+        metrics.update(
+            {
+                "WORLD_MODEL_learned_initial_h": (
+                    self.module[module_id].world_model.initial_h
+                ),
+                # Prediction losses.
+                # Decoder (obs) loss.
+                "WORLD_MODEL_L_decoder_B_T": prediction_losses["L_decoder_B_T"],
+                "WORLD_MODEL_L_decoder": prediction_losses["L_decoder"],
+                # Reward loss.
+                "WORLD_MODEL_L_reward_B_T": prediction_losses["L_reward_B_T"],
+                "WORLD_MODEL_L_reward": prediction_losses["L_reward"],
+                # Continue loss.
+                "WORLD_MODEL_L_continue_B_T": prediction_losses["L_continue_B_T"],
+                "WORLD_MODEL_L_continue": prediction_losses["L_continue"],
+                # Total.
+                "WORLD_MODEL_L_prediction_B_T": prediction_losses["L_prediction_B_T"],
+                "WORLD_MODEL_L_prediction": prediction_losses["L_prediction"],
+                # Dynamics loss.
+                "WORLD_MODEL_L_dynamics_B_T": L_dyn_B_T,
+                "WORLD_MODEL_L_dynamics": L_dyn,
+                # Representation loss.
+                "WORLD_MODEL_L_representation_B_T": L_rep_B_T,
+                "WORLD_MODEL_L_representation": L_rep,
+                # Total loss.
+                "WORLD_MODEL_L_total_B_T": L_world_model_total_B_T,
+                "WORLD_MODEL_L_total": L_world_model_total,
+            }
+        )
+        # And register all the computed metrics.
+        self.register_metrics(module_id=module_id, metrics_dict=metrics)
+
+        # Return the total loss as a sum of all individual losses.
+        return (
+            L_world_model_total
+            + critic_loss_results["CRITIC_L_total"]
+            + actor_loss_results["ACTOR_L_total"]
+            if hps.train_actor
+            else 0.0
+        )
+
+    def _compute_world_model_prediction_losses(
+        self,
+        *,
+        hps,
+        rewards,
+        continues,
+        fwd_out,
+    ) -> ResultDict:
+        """Helper method computing all world-model related prediction losses.
+
+        Prediction losses are those to train the predictors of the world model:
+        Reward predictor, continue predictor, and the decoder (which predicts
+        observations).
+        """
+
+        # Learn to produce symlog'd observation predictions.
+        # If symlog is disabled (e.g. for uint8 image inputs), `obs_symlog_BxT` is the
+        # same as `obs_BxT`.
+        obs_BxT = fwd_out["sampled_obs_symlog_BxT"]
+        obs_distr = fwd_out["obs_distribution_BxT"]
+        # Fold time dim and flatten all other (image?) dims.
+        obs_BxT = tf.reshape(
+            obs_BxT, shape=[-1, int(np.prod(obs_BxT.shape.as_list()[1:]))]
+        )
```

Comment:
Would this also work:
obs_BxT = tf.reshape(obs_BxT, shape=[-1, tf.reduce_prod(obs_BxT.shape[1:])])
?

Reply:
Yeah, let's try this. Seems cleaner :)